chain/client/src/sync/header.rs

```diff
@@ -89,11 +89,19 @@ impl HeaderSync {
                 Some(height) => height,
                 None => chain.head()?.height,
             };
-            *sync_status = SyncStatus::HeaderSync {
-                start_height,
-                current_height: header_head.height,
-                highest_height,
-            };
+
+            // An artificial span to easily detect a node getting into the sync state in Grafana.
+            {
+                let _span =
+                    tracing::debug_span!(target: "sync", "set_sync", sync = "HeaderSync").entered();
+                tracing::debug!(target: "sync", prev_sync_status = ?sync_status);
+                *sync_status = SyncStatus::HeaderSync {
+                    start_height,
+                    current_height: header_head.height,
+                    highest_height,
+                };
+                tracing::debug!(target: "sync", ?sync_status);
```

Comment:
Any reason these events are not part of the same span? Outputting two events in sequence seems to be strictly more difficult to reason about, than a single event with all data in it, no? (And then the artificial span wouldn’t be necessary either…)

(If you do end up rewriting this, evaluate if [recording fields after the fact](https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.record) into the top-level `run` span is applicable)

Reply:
wow, I haven't seen `field::Empty` before, thanks.
Single event is much simpler than multiple events, yes.

But I do need the artificial span to easily detect when a node goes into the sync mode. There are about 5 different places all in different functions that change the sync mode.

Maybe I need a helper function, and then I'll sneak the span into that helper function.